deepinv/models/flowunet.py

```diff
@@ -0,0 +1,523 @@
+# Unet copied from "A Variational Perspective on Diffusion-Based Generative Models and Score Matching", NeurIPS 2021, Huang et al
+# https://github.com/CW-Huang/sdeflow-light
+# which is modified from the "Denoising Diffusion Probabilistic Models", Ho et al
+
+import math
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from torch.nn.init import _calculate_fan_in_and_fan_out
+
+
+from .base import Denoiser
+
+
+def get_weights_from_drive(pretrained_name):
+    if pretrained_name == "celeba":
+        drive_id = "1ZZ6S-PGRx-tOPkr4Gt3A6RN-PChabnD6"
+    elif pretrained_name == "afhq_cat":
+        drive_id = "1FpD3cYpgtM8-KJ3Qk48fcjtr1Ne_IMOF"
+    return f"https://drive.google.com/uc?id={drive_id}"
+
+
+class Swish(nn.Module):
+    def __init__(self):
+        super().__init__()
+
+    def forward(self, x):
+        return torch.sigmoid(x) * x
+
+
+def group_norm(out_ch):
+    return nn.GroupNorm(num_groups=32, num_channels=out_ch, eps=1e-6, affine=True)
+
+
+# TODO: change init
+
+
+class FlowUNet(Denoiser):
+    def __init__(
+        self,
+        input_channels,
+        input_height,
+        ch=32,
+        output_channels=None,
+        ch_mult=(1, 2, 4, 8),
+        num_res_blocks=6,
+        attn_resolutions=(16, 8),
+        dropout=0.0,
+        resamp_with_conv=True,
+        act=Swish(),
```

Comment:
@matthieutrs how do you feel wrt having mutable parameters as defaults? SHould we do act=None and then `self.act = Swish() if act is None else act`